Event/AuthorizationRequestResolveEvent.php

```diff
@@ -0,0 +1,115 @@
+<?php
+
+namespace Trikoder\Bundle\OAuth2Bundle\Event;
+
+use League\OAuth2\Server\Entities\ClientEntityInterface;
```

Comment:
@ajgarlag We should make the return values consistent with existing event objects. In other words, return bundle specific entities instead of the ones provided by League. This also includes targeting `UserInterface` provided by Symfony.

Examples:

* https://github.com/trikoder/oauth2-bundle/blob/master/Event/ScopeResolveEvent.php#L60
* https://github.com/trikoder/oauth2-bundle/blob/master/Event/UserResolveEvent.php#L70

Reply:
Not sure about this. To address this, I should change at least 4 methods:
```php
final class AuthorizationRequestResolveEvent extends Event
{
//...
    public function getClient(): ClientEntityInterface
    {
        return $this->authorizationRequest->getClient();
    }

    public function getUser(): UserEntityInterface
    {
        return $this->authorizationRequest->getUser();
    }

    public function setUser(UserEntityInterface $user): void
    {
        $this->authorizationRequest->setUser($user);
    }

    /**
     * @return ScopeEntityInterface[]
     */
    public function getScopes(): array
    {
        return $this->authorizationRequest->getScopes();
    }

/...
}
```
* The `getScopes` method could use the `ScopeConverter` helper to transform league entities into bundle models, so no problem here.
* The `setUser` method could receive a Symfony `UserInterface` instance an transform it into a League `UserEntityInterface` before setting it into the `AuthorizationRequest` object.
* With the `getUser` method I do not see how to transform the `UserEntityInterface` from the `AuthorizationRequest` into a Symfony `UserInterface`. We could store the instance received from the `setUser` method in a private property, but if the `AuthorizationRequest` received as constructor parameter already has an user set, how can we transform it into an Syfmony `UserInterface` to expose this info to the listeners?
* Finally, in the `getClient` method, we need to have a reference to a `ClientManagerInterface` instance to find the `Model\Client` using the `ClientEntityInterface` identifier.

WDYT?

Reply:
@ajgarlag I took a look at League's implementation and they don't seem to set the user themselves anywhere in the library. This is the only place where the `AuthorizationRequest` gets instantiated:

https://github.com/thephpleague/oauth2-server/blob/7.4.0/src/Grant/AuthCodeGrant.php#L272

With that being said, I think we can still remain consistent regarding `UserInterface`. Here are my suggestions:

* Change the `AuthorizationRequestResolveEvent` event so it doesn't directly manipulate the `AuthorizationRequest` object. Events should only communicate the desired state. We can set the user on `AuthorizationRequest` from where the event gets dispatched (the controller).
* If multiple services are required to construct the `AuthorizationRequestResolveEvent`, we can move it to a dedicated class.
* It seems we have a need to convert the entities yet again. Perhaps it's time to introduce a class similar to the `ScopeConverter` so we avoid code duplication. Here's what I'm [referring to](https://github.com/trikoder/oauth2-bundle/blob/master/League/Repository/UserRepository.php#L61).

These changes would also make `AuthorizationRequestUserResolvingListener` way simpler.

Reply:
>  * Change the AuthorizationRequestResolveEvent event so it doesn't directly manipulate the AuthorizationRequest object. Events should only communicate the desired state. We can set the user on AuthorizationRequest from where the event gets dispatched (the controller).

Done.
>  * If multiple services are required to construct the AuthorizationRequestResolveEvent, we can move it to a dedicated class.

Done, but I'm very bad at naming things, so maybe the event factory should be renamed and/or relocated.
>  * It seems we have a need to convert the entities yet again. Perhaps it's time to introduce a class similar to the ScopeConverter so we avoid code duplication. Here's what I'm referring to

Done.

Reply:
@ajgarlag Looking good! 👍